### ISSUE TITLE
File Upload with Swagger

### ISSUE THREAD

**Issue by amberream** [stoicflame/enunciate#480]
I submitted a similar issue a few months ago (#383) that is fixed in 2.4.  But the solution doesn't fix all cases where a file input is expected.  Here is my method signature:

``` java
        @PUT
        @Consumes({ PDF, GIF, JPEG, PNG, TIFF, MediaType.TEXT_PLAIN, DOC, DOCX })
        @TypeHint(DocumentModel.class)
        public Response put(InputStream file,@DefaultValue(PDF) @HeaderParam("Content-Type") String contentType, @DefaultValue("72") @QueryParam("dpi") Integer dpi)
```

In the swagger docs I'd expect to see a file input for the "file" parameter.  But instead I just get a text area.

**Comment by stoicflame** [comment]
It's because of that `text/plain` media type. The algorithm returns a text box because the text-based media types (including XML, JSON) are the most common case. Can you suggest a better algorithm that doesn't subvert the common case?

**Comment by amberream** [comment]
I haven't looked through the enunciate code, but I would think the parameter type (InputStream) would be the most relevant thing to look at.  Can you read the method signatures or just the annotations?

**Comment by stoicflame** [comment]
Method signatures are available, but just because the parameter is an `InputStream` doesn't mean it's not text-based. You can take an `InputStream` even for XML or JSON. It happens a lot in cases where you'd like to use stream-based parsing instead of putting everything into a DOM or and object model.

**Comment by stoicflame** [comment]
I figured out a way to set a "preferred" media type using the quality-of-source parameter at b52fdbc. If you use the `qs` to prioritize PDF (or whatever) above `text/plain`, then the file-based input will be used instead of the text, e.g.:

```java
@PUT
@Consumes({ PDF + "; qs=2.0", GIF, JPEG, PNG, TIFF, MediaType.TEXT_PLAIN, DOC, DOCX })
@TypeHint(DocumentModel.class)
public Response put(InputStream file,@DefaultValue(PDF) @HeaderParam("Content-Type") String contentType, @DefaultValue("72") @QueryParam("dpi") Integer dpi)
```

**Comment by stoicflame** [comment]
[Enunciate 2.6 has been released.](https://github.com/stoicflame/enunciate/releases/tag/v2.6.0)

**Comment by amberream** [comment]
When I set the qs value as you showed in your comment above I do get the upload button as expected.  This method accepts, pdf, png, jpeg, etc, but when I try to upload a file I see the following error in the js console when I click the "try it out button" on the swagger page:
`server doesn't consume multipart/form-data, try ["image/png","application/msword","application/pdf","image/jpeg","application/vnd.openxmlformats-officedocument.spreadsheetml.sheet","image/tiff","image/gif","application/vnd.openxmlformats-officedocument.wordprocessingml.document","text/plain"]`

**Comment by stoicflame** [comment]
I'm not sure why Swagger is trying to upload `multipart/form-data`. Can you provide a change to the `swagger.json` that will fix this issue?

**Comment by stoicflame** [comment]
Moving fix to 2.13, pending more information.

**Comment by stoicflame** [comment]
Still no details about this issue. I'll close it for now. I'm happy to re-open when more information is provided.
